app/src/main/java/com/jakdor/apapp/ui/MainActivity.kt

```diff
@@ -29,14 +47,72 @@ class MainActivity : AppCompatActivity(), HasSupportFragmentInjector {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        val externalStorageCheck = ContextCompat.checkSelfPermission(this,
+            Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        val cameraCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA)
+
+        if (externalStorageCheck != PackageManager.PERMISSION_GRANTED &&
+            cameraCheck != PackageManager.PERMISSION_GRANTED) {
+            Nammu.askForPermission(this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                Manifest.permission.CAMERA), object : PermissionCallback {
+                override fun permissionGranted() {
+
+                }
+
+                override fun permissionRefused() {
+                    finish()
+                }
+
+            })
+        }
+
+            options = Options.init()
+                .setRequestCode(100)
+                .setCount(8)
+                .setPreSelectedUrls(returnedImages)
+                .setPath("/DCIM/TrueHome")
+
+        //switchToAddApartmentFragment()
         if(authRepository.isLoggedIn()){
-            switchToApartmentListFragment()
+            switchToAddApartmentFragment()
         }
         else{
             switchToLoginFragment()
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        val apartmentFragment = supportFragmentManager.findFragmentByTag(ApartmentFragment.CLASS_TAG) as ApartmentFragment
```

Comment:
A co jak fragmentu nie ma? Btw. Co w przypadku jak będziemy chcieli dodawać zdjęcia także w innych miejscach apki? Tak jak teraz jest to jest bardzo mocno nie skalowalne i nie testowalne tak przy okazji też.

Reply:
Od biedy na razie by mogło tak zostać, ale to na pewno jest do poprawy